File: DSA/Math/0009_Palindrome_Number/php/solution_2.php
// 0009. Palindrome Number
// Difficulty : Easy
// Tags       : math
// Solved on  : 2026-04-21
// Attempt    : #2
// LeetCode   : https://leetcode.com/problems/palindrome-number/

class Solution {

    /**
     * @param Integer $x
     * @return Boolean
     */
    function isPalindrome($x) {
        return $x >= 0 && (string)$x == strrev((string)$x);
    }
}

File: Programming Languages/CPP/0657_Robot_Return_to_Origin/solution_2.cpp
// 0657. Robot Return to Origin
// Difficulty : Easy
// Tags       : string, simulation
// Solved on  : 2019-07-26
// Attempt    : #2
// LeetCode   : https://leetcode.com/problems/robot-return-to-origin/

class Solution {
public:
    bool judgeCircle(string moves) {
        int x=0,y=0;
        for(int i=0;i<moves.length();i++){
            if(moves[i]=='U'){
                y++;
            }else if(moves[i]=='D'){
                y--;
            }else if(moves[i]=='L'){
                x--;
            }else if(moves[i]=='R'){
                x++;
            }
        }
        if(x==0 && y==0){
            return true;
        }else{
            return false;
        }
    }
};